### PR TITLE
Test for SapiEmitter

### PR DESCRIPTION
**Pull request by zhukovra**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ❌
| Tests pass?   | ✔️
| Fixed issues  | #105

### CODE REVIEW

File: tests/Emitter/SapiEmitterTest.php [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+        $this->assertEquals(200, http_response_code());
+        $this->checkHeadersEquals([
+            'X-Test: 1',
+            'Content-Length: ' . strlen($body),
+        ]);
+        $this->assertEquals($body, $result);
+    }
+
+    public function testEmptyBodyOnCode()
```
Comment (samdark) [inline tests/Emitter/SapiEmitterTest.php]:
```suggestion
    /**
     * @test
     */
    public function shouldNotOutputBodyWhenResponseCodeIs204()
```

File: tests/Emitter/SapiEmitterTest.php [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        ob_end_clean();
+
+        $this->assertEquals(204, http_response_code());
+        $this->checkHeadersEquals([
+            'X-Test: 1',
+        ]);
+        $this->assertEmpty($result);
+    }
+
+    public function testEmptyBodyOnFlag()
```
Comment (samdark) [inline tests/Emitter/SapiEmitterTest.php]:
```suggestion
    /*
     * @test
     */
    public function shouldNotOutputBodyIfEmitToldSo()
```

File: tests/Emitter/SapiEmitterTest.php [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        ob_end_clean();
+
+        $this->assertEquals(200, http_response_code());
+        $this->checkHeadersEquals([
+            'X-Test: 1',
+        ]);
+        $this->assertEmpty($result);
+    }
+
+    public function testContentLengthHeader()
```
Comment (samdark) [inline tests/Emitter/SapiEmitterTest.php]: What does it test compared to testEmit()?

Comment (zhukovra) [reply tests/Emitter/SapiEmitterTest.php]: Check what emit() doesn't compute `Content-length` header: 
https://github.com/yiisoft/yii-web/blob/5db3424fefb3a377da1d74ceae80916691b1ac5e/src/Emitter/SapiEmitter.php#L40

Comment (samdark) [reply tests/Emitter/SapiEmitterTest.php]:
```suggestion
    /*
     * @test
     */
    public function contentLengthShouldNotBeOverwrittenIfPresent()
```

**Comment by samdark** [comment]
Merged. Thank you!

**Pull request merged**